data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template

```diff
@@ -26,8 +26,11 @@ for name in ironic ironic-inspector ironic-ramdisk-logs dnsmasq httpd coreos-dow
 done
 
 # Start the provisioning nic if not already started
-PROVISIONING_NIC={{.PlatformData.BareMetal.ProvisioningInterface}}
 
+#Retrieve the provisioning interface name based on mac address.
+PROVISIONING_NIC_MAC={{.PlatformData.BareMetal.ProvisioningInterfaceMAC}}
+PROVISIONING_NIC=$(ip -o link | grep -i $PROVISIONING_NIC_MAC | awk '{print $2}' | sed 's/://g')
```

Comment:
@zaneb wanted to catch when a user used capital/lowercase in the MAC override, and combined into one line: 
```
PROVISIONING_NIC=$(ip -j -o link | jq -r '.[] | select(.address | match ("{{.PlatformData.BareMetal.ProvisioningInterfaceMAC}}";"i")).ifname')
```

Reply:
The `${PROVISIONING_NIC_MAC,,[A-F]}` converts to lowercase, but I like the idea of combining into one line. However, running that command on my laptop I get a non-zero return code and this error:

```
jq: error (at <stdin>:1): null (null) cannot be matched, as it is not a string
```

because there's an interface (`tun0`) without a MAC address.

Reply:
ah, didn't take that use case into account. This worked for me on my local and bootstrapvm: 
`ip -j -o link | jq -r '.[] | select(.address != null) | select(.address | match ("{{.PlatformData.BareMetal.ProvisioningInterfaceMAC}}";"i")).ifname'`

Reply:
Hey @zaneb are you good with this change? Happy to refactor if you think it needs it.

Reply:
The latest one looks fine to me.